### PR TITLE
Implement ASSERT statement

### PR DESCRIPTION
**Pull request by Dandandan**
Issue: https://github.com/ballista-compute/sqlparser-rs/issues/225

See:
https://cloud.google.com/bigquery/docs/reference/standard-sql/debugging-statements
https://www.postgresql.org/docs/10/plpgsql-errors-and-messages.html

PostgreSQL supports an arbitrary expression as optional second message, so made that an expression as well.

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -551,6 +551,14 @@ pub enum Statement {
     Rollback { chain: bool },
     /// CREATE SCHEMA
     CreateSchema { schema_name: ObjectName },
+
+    /// ASSERT <condition> [AS <message>]
+    Assert {
+        condition: Box<Expr>,
```
[the remark this answers is not in the document]
Comment (Dandandan) [reply src/ast/mod.rs]: Removed! Probably doesn't matter much in this case, and makes it easier to read. Would maybe be interesting in the future to see how much bytes we are using per struct (and impact on perf / memory).

**Comment by Dandandan** [comment]
Thanks!

File: src/ast/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickolay) [reply src/ast/mod.rs]: I assume that the only two cases when we should use `Box` in our AST are: cycles (e.g. `Expr`) and when clippy reports a [large_enum_variant](https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant).

[the remark this answers is not in the document]
Comment (Dandandan) [reply src/ast/mod.rs]: That's probably a good heuristic indeed.

**Pull request merged**